test_runner/src/test/kotlin/TmpV2.kt

```diff
@@ -0,0 +1,64 @@
+import com.google.api.client.json.GenericJson
+import ftl.args.AndroidArgs
+import ftl.gc.GcTestMatrix
+import ftl.reports.api.createJUnitTestResult
+import ftl.reports.api.createTestExecutionDataListAsync
+import ftl.reports.api.data.TestExecutionData
+import ftl.reports.api.filterForJUnitResult
+import ftl.reports.xml.xmlToString
+import java.io.File
+import kotlin.system.exitProcess
+
+object TmpV2 {
+    private const val PREFIX = "REPORT_API_TEST"
+    private const val MATRIX_ID = "matrix-2ut05apux3ugd"
+    private const val JUNIT_REPORT_FILE = "$PREFIX-JUnitReport.xml"
+    private const val API_JSON_FILE = "$PREFIX-api-result.json"
+
+    @JvmStatic
+    fun main(args: Array<String>) {
+
+        print("fetching matrix")
+        val matrix = GcTestMatrix.refresh(MATRIX_ID, AndroidArgs.default())
+        println(" - OK")
+
+        print("generating api results")
+        val apiResult = matrix.testExecutions
+            .createTestExecutionDataListAsync()
+            .filterForJUnitResult()
+            .createJson()
+        println(" - OK")
+
+        print("writing api results to file")
+        File(API_JSON_FILE).writeText(apiResult.toPrettyString())
+        println(" - OK")
+
+        print("generating junit report")
+        val jUnitTestResult = matrix.createJUnitTestResult()
+        println(" - OK")
+
+        print("writing api results to file")
+        File(JUNIT_REPORT_FILE).writeText(jUnitTestResult.xmlToString())
```

Comment:
To test this, run an Android app with multiple tests using server side sharding that has flaky tests. Use `num-flaky-test-attempts` set to a non-zero value such as 2. Then compare the JUnit XML file that FTL is creating vs the one we create via the API.

Reply:
I was testing exactly that case. I know there is a difference between old and new JUnitReport.xml. 
Old: 
```
<?xml version='1.0' encoding='UTF-8' ?>
<testsuites>
  <testsuite name="NexusLowRes-28-en-portrait#" tests="3" failures="0" errors="0" skipped="0" time="3.103" timestamp="2020-03-10T17:12:24" hostname="localhost">
    <testcase name="test0" classname="com.example.test_app.InstrumentedTest" time="1.161">
      <webLink></webLink>
    </testcase>
    <testcase name="test1" classname="com.example.test_app.InstrumentedTest" time="1.034">
      <webLink></webLink>
    </testcase>
    <testcase name="test2" classname="com.example.test_app.InstrumentedTest" time="0.908">
      <webLink></webLink>
    </testcase>
  </testsuite>
</testsuites>
```
New:
```
<?xml version='1.0' encoding='UTF-8' ?>
<testsuites>
  <testsuite name="NexusLowRes-28-en-portrait" tests="1" failures="0" errors="0" skipped="0" time="1.161" timestamp="2020-03-10T17:07:54.422Z">
    <testcase name="test0" classname="com.example.test_app.InstrumentedTest" time="1.161"/>
  </testsuite>
  <testsuite name="NexusLowRes-28-en-portrait" tests="2" failures="0" errors="0" skipped="0" time="1.942" timestamp="2020-03-10T17:07:54.422Z">
    <testcase name="test1" classname="com.example.test_app.InstrumentedTest" time="1.034"/>
    <testcase name="test2" classname="com.example.test_app.InstrumentedTest" time="0.908"/>
  </testsuite>
</testsuites>
```

1. In new JUnitReport suites reflects shards. I can fix it easily, but that was first what I achieved. I guess that maybe is better to expose some info about shards in JUnitReport. If you don't like that I will fix it.achieved 

2. The `timestamp` is different yes. Old timestamp is simply timestamp of first of downloaded reports. New one is not correct I will fix it.

3. I need to also fix weblinks.

Reply:
makes sense!